runtime/src/accounts_db.rs

```diff
@@ -463,7 +447,8 @@ impl AccountsDB {
                     fork_id, store.fork_id,
                     "AccountDB::accounts_index corrupted. Storage should only point to one fork"
                 );
-                store.remove_account();
+                let mut count_and_status = store.count_and_status.write().unwrap();
```

Comment:
why pull the lock out here?

Reply:
Heh, I was trying to convey to future callers that the count_and_status() lock must be held for the entire duration of the function. I did the same for `set_status()`, but that one actually requires the change b/c the lock is held by the time it's called. Here, I guess I could revert this  change and just add a comment?

Reply:
ah, I see.  I dunno, is ugly either way.

Reply:
I like the idea of a "try_acquire()" for the fork_storage() function, though..

Reply:
and that would give us automatic parallelism on append_accounts(), because contending writes to a candidates would cause creation of new candidates.

Reply:
haha I should've refreshed for comments, I think we're on the same page xD

Reply:
https://github.com/solana-labs/solana/pull/4523